### PR TITLE
Zaowang/liftr nginx/adding 2 24 09 01 preview

### PR DESCRIPTION
**Pull request by zaowang-ms**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Comment by zaowang-ms** [comment]
Note: Just duplicating and publishing changes from: https://github.com/Azure/azure-rest-api-specs-pr/pull/19071

**Comment by mikekistler** [comment]
Breaking changes previously reviewed and approved in https://github.com/Azure/azure-rest-api-specs-pr/pull/19071

**Comment by zaowang-ms** [comment]
The CI lint diff check contradicts to our 
[previous conversation in the private PR](https://github.com/Azure/azure-rest-api-specs-pr/pull/19071#discussion_r1744229900). We've already implemented our code that way and we hope to publish the API in consistence with the private API repo. Please suppress it as we're hoping to get it merged soon. Thanks!

**Comment by zaowang-ms** [comment]
Hi, 
 
Please verify my understanding. From what I see, the API change in the private repo follows your suggestion on separating request/response objects given secret data is passed from requests but not returned by responses (remove the secretText field in response)
https://github.com/Azure/azure-rest-api-specs-pr/pull/19071#discussion_r1744234166
 
The PR was approved in private repo several weeks ago. We've rollout the ARM manifest and have the API live and tested in production as well. The new portal blade that uses this API is also in preview. 
 
The nature of this API is asymmetric in terms of request/response data given secret is write-only and the returned hash is read-only. So whether modeling request/response via the same entity or separate, either one works fine from user perspective as long as the annotation clarifies it. From this perspective, the lintdiff warning seems to more of an artificial side effect. 
 
Revisiting and changing the API at this point will require changing the RP implementation, redeploying RP, changing the portal implementation alongside with extra time for API review in both private/public repos. And hence the estimated 2 weeks delay. We plan to demonstrate the feature in Ignite.
 
I'm wondering if we can expedite the public repo merge through an exception?

**Comment by ramoka178** [comment]
> Hi,
> 
> Please verify my understanding. From what I see, the API change in the private repo follows your suggestion on separating request/response objects given secret data is passed from requests but not returned by responses (remove the secretText field in response) [Azure/azure-rest-api-specs-pr#19071 (comment)](https://github.com/Azure/azure-rest-api-specs-pr/pull/19071#discussion_r1744234166)
> 
> The PR was approved in private repo several weeks ago. We've rollout the ARM manifest and have the API live and tested in production as well. The new portal blade that uses this API is also in preview.
> 
> The nature of this API is asymmetric in terms of request/response data given secret is write-only and the returned hash is read-only. So whether modeling request/response via the same entity or separate, either one works fine from user perspective as long as the annotation clarifies it. From this perspective, the lintdiff warning seems to more of an artificial side effect.
> 
> Revisiting and changing the API at this point will require changing the RP implementation, redeploying RP, changing the portal implementation alongside with extra time for API review in both private/public repos. And hence the estimated 2 weeks delay. We plan to demonstrate the feature in Ignite.
> 
> I'm wondering if we can expedite the public repo merge through an exception?

Going by the previous private PR's approval. Could you add a suppression for this issues in readme?

**Comment by zaowang-ms** [comment]
@ramoka178 thanks for taking a look! I don't know how to do it, could you share a link / doc on how to do this?

**Comment by ramoka178** [comment]
> @ramoka178 thanks for taking a look! I don't know how to do it, could you share a link / doc on how to do this?

@zaowang-ms  You could check any other swagger's read me that has suppressions. 
e.g. ( no preference, just searched suppressions on the repo ) https://github.com/Azure/azure-rest-api-specs/blob/362baa4049e19f5a01c3d18760e17f5f22583276/specification/oep/resource-manager/readme.md?plain=1#L51

**Comment by zaowang-ms** [comment]
@ramoka178 added the suppression, please review. Thanks a lot!

File: specification/nginx/resource-manager/readme.md [hunk 1]
```diff
@@ -96,6 +96,15 @@ input-file:
 - NGINX.NGINXPLUS/preview/2021-05-01-preview/swagger.json
 ```
 
+## Suppression
+``` yaml
+directive:
+  - suppress: PutRequestResponseSchemeArm
+    from: swagger.json
+    reason: Temporary suppression needed to avoid delays for business needs and maintain production timelines
```
Comment (ramoka178) [inline specification/nginx/resource-manager/readme.md:104]: Please update that this is previously approved

Comment (zaowang-ms) [reply specification/nginx/resource-manager/readme.md]: Updated!

**Review by jingwenw15 — APPROVED** [review]

**Pull request merged**